### ISSUE TITLE
Question about overflow: auto in the spec

### ISSUE THREAD

**Issue by yawpitch** [ampproject/amphtml#21042]
The [format specification](https://github.com/ampproject/amphtml/blob/master/spec/amp-html-format.md) states that:

> `overflow` (and `overflow-y`, `overflow-x`) may not be styled as “auto” or “scroll”. No user defined element in an AMP document may have a scrollbar.

However *many* sites that purport to be AMP HTML and which are considered "Valid AMP" by the Validator -- including both [amproject.org](https://www.ampproject.org/) and [ampbyexample.com](https://ampbyexample.com/) -- use overflow-x: auto in `<code>` elements.

Is the above quote out of data and the specification _does_ allow overflowing elements? Or is the validator broken and large numbers of seemingly valid sites are not in fact valid AMP?

**Comment by aghassemi** [comment]
Hi @yawpitch 

This is great question with a big of complicated answer:

So far and currently (and until https://github.com/ampproject/amphtml/issues/14088 lands) behaviour of any sub-scrollable areas in AMP is buggy. It may or may not work depending on how it is used. For example:

1- Vertical sub-scrollers:  If they are any `amp-*` elements that are not visible, they may not become visible after user scrolls the sub area. If it is just text, it is fine. Also scrollale sub-areas is against our UX principles on mobile (desktop is fine)
2- Horizontal sub-scroller: same issue as above and additionally they break swipe beheavior in the viewer. solution, use `<amp-carousel type=carousel>`

We did actually wanted to enforce these rules that you mentioned through validation but at the time we were not doing CSS parsing for validation and did not enforce it per spec. We can't enforce now since it would be a big breacking change.

tl;dr;: avoid hem if you can and use the alternatives mentioned above.

I hope this helps.

**Comment by yawpitch** [comment]
Thanks for the explanation. My need is the same as on ampbyexample, <pre> and <code> blocks demonstrating source code snippets. Text only, no amp elements; truncating the overflow is not an option, and neither is breaking the lines. The use of overflow-x: auto is essentially ubiquitous for this case, while amp-carousel seems to be an unfamiliar and unpleasant interface alternative for that purpose.

 Thank you though. M
